packages/javascript/bh-shared-ui/src/components/AnalyzeNowConfiguration/AnalyzeNowConfiguration.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright 2025 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+import { Button } from '@bloodhoundenterprise/doodleui';
+import { RequestOptions } from 'js-client-library';
+import { useState } from 'react';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import { useNotifications } from '../../providers';
+import { apiClient } from '../../utils';
+import AnalyzeNowConfirmDialog from './AnalyzeNowConfirmDialog';
+
+const requestAnalysis = (options?: RequestOptions) => {
+    return apiClient.requestAnalysis(options).then((res) => res.data);
+};
+
+const useRequestAnalysis = () => {
+    const queryClient = useQueryClient();
+    return useMutation(requestAnalysis, {
+        onSuccess: () => {
+            queryClient.invalidateQueries('datapipe-status');
+        },
+    });
+};
+
+type AnalyzeNowProps = {
+    description: string;
+    note?: string;
+};
+
+const AnalyzeNowConfiguration: React.FC<AnalyzeNowProps> = ({ description, note }) => {
+    const [isOpenDialog, setIsOpenDialog] = useState(false);
+    const requestAnalysis = useRequestAnalysis();
+
+    const { addNotification } = useNotifications();
+
+    const showDialog = () => setIsOpenDialog(true);
+    const hideDialog = () => setIsOpenDialog(false);
```

Comment:
After the suggestion above, you can remove the notifications hook here. The  analysis hook will need to come after the dialog hide method so you can pass it in. Also, it's a useful pattern to destructure the mutate method and alias it.

```suggestion
    const showDialog = () => setIsOpenDialog(true);
    const hideDialog = () => setIsOpenDialog(false);
    
    const { mutate: requestAnalysis } = useRequestAnalysis(hideDialog);
```